cmake/metadata.cmake

```diff
@@ -100,7 +102,7 @@ set(events_src_path "${PX4_SOURCE_DIR}/src/lib/events")
 add_custom_target(metadata_extract_events
 	COMMAND ${CMAKE_COMMAND} -E make_directory ${PX4_BINARY_DIR}/events
 	COMMAND ${PYTHON_EXECUTABLE} ${PX4_SOURCE_DIR}/Tools/px_process_events.py
-		--src-path ${PX4_SOURCE_DIR}/src
+		--src-path ${all_px4_src_files}
```

Comment:
We need to make sure to add all modules here, but we could exclude the unit tests.
Simpler yet, you can add the tag to the failing case:
```
		/* EVENT
		 */
		reporter.armingCheckFailure(NavModes::All, health_component_t::remote_control,
					    events::ID("arming_test_basic_fail_all_modes_fail1"), events::Log::Info, "");
```

Reply:
That's still not working, what do you think is the underlying problem?

``` Console
[0/2] Re-checking globbed directories...
[0/1] Extracting events from full source
Exception while parsing file /home/dagar/git/PX4-Autopilot/src/modules/commander/HealthAndArmingChecks/HealthAndArmingChecksTest.cpp
Exception in file /home/dagar/git/PX4-Autopilot/src/modules/commander/HealthAndArmingChecks/HealthAndArmingChecksTest.cpp
Traceback (most recent call last):
  File "/home/dagar/git/PX4-Autopilot/Tools/px_process_events.py", line 102, in <module>
    main()
  File "/home/dagar/git/PX4-Autopilot/Tools/px_process_events.py", line 89, in main
    if not scanner.ScanDir(src_paths, parser):
  File "/home/dagar/git/PX4-Autopilot/Tools/px4events/srcscanner.py", line 28, in ScanDir
    if not self.ScanFile(path, parser):
  File "/home/dagar/git/PX4-Autopilot/Tools/px4events/srcscanner.py", line 49, in ScanFile
    return parser.Parse(contents)
  File "/home/dagar/git/PX4-Autopilot/Tools/px4events/srcparser.py", line 249, in Parse
    event.validate()
  File "/home/dagar/git/PX4-Autopilot/Tools/px4events/srcparser.py", line 83, in validate
    assert self.message[-1] != '.', "Avoid event message ending in '.' ({:})".format(self.message)
IndexError: string index out of range
FAILED: CMakeFiles/metadata_extract_events /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/CMakeFiles/metadata_extract_events 
cd /home/dagar/git/PX4-Autopilot/build/px4_sitl_default && /home/dagar/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E make_directory /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/events && /usr/bin/python3 /home/dagar/git/PX4-Autopilot/Tools/px_process_events.py --src-path /home/dagar/git/PX4-Autopilot/src --json /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/events/px4_full.json && /usr/bin/python3 /home/dagar/git/PX4-Autopilot/src/lib/events/libevents/scripts/combine.py /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/events/px4_full.json /home/dagar/git/PX4-Autopilot/src/lib/events/libevents/events/common.json /home/dagar/git/PX4-Autopilot/src/lib/events/enums.json --output /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/events/all_events_full.json && /usr/bin/python3 /home/dagar/git/PX4-Autopilot/src/lib/events/libevents/scripts/validate.py /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/events/all_events_full.json && /usr/bin/python3 /home/dagar/git/PX4-Autopilot/Tools/compress.py /home/dagar/git/PX4-Autopilot/build/px4_sitl_default/events/all_events_full.json
ninja: build stopped: subcommand failed.
make[1]: *** [Makefile:232: px4_sitl_default] Error 1
make: *** [Makefile:361: extract_events] Error 2
```

Reply:
I pushed a fix to https://github.com/PX4/PX4-Autopilot/tree/pr-cmake_metadata_extract_events_skip_file